### PR TITLE
Parser changes for the new ruby model

### PR DESCRIPTION
**Pull request by darobin**
As per http://darobin.github.io/html-ruby/. This change is being landed in the W3C specification, apply with caution.

### CODE REVIEW

File: html5lib/html5parser.py [hunk 1]
```diff
@@ -901,7 +901,7 @@ def __init__(self, parser, tree):
                 ("iframe", self.startTagIFrame),
                 (("noembed", "noframes", "noscript"), self.startTagRawtext),
                 ("select", self.startTagSelect),
-                (("rp", "rt"), self.startTagRpRt),
+                (("rp", "rt", "rb", "rtc"), self.startTagRpRtRbRtc),
```
Comment (annevk) [inline html5lib/html5parser.py]: This is wrong. You need two distinct groups. Or you need to also branch the `self.tree.openElements[-1].name != "ruby"` line further. What you're doing now doesn't match other implementations.

Comment (darobin) [inline html5lib/html5parser.py]: I'm not sure that this PR is up to date with the more recent changes to the spec anyway.

Comment (annevk) [inline html5lib/html5parser.py]: It isn't.

**Comment by gsnedders** [comment]
I'm just going to close this as outdated and deal with this with the larger spec update.

**Pull request closed without merging**